Fixed OS X i386 Xcode builds on x86_64 build machines

HofiOne opened the pull request:
1.)
Without assigning App-OSX.xcconfig to Kodi.app target building i386
target will not fix dependent libraries when bundling the Kodi.app with
copyframeworks-osx.command

copyframeworks-osx.command uses XBMC_DEPENDS that composed like
$(XBMC_DEPENDS_ROOT)/$(SDK_NAME)_$(CURRENT_ARCH)-target

CURRENT_ARCH not set correctly, without assigning App-OSX.xcconfig the
ONLY_ACTIVE_ARCH is not set and the builder machine architecture will
be used instead of the targeted architecture.

2.)
Also found that XCode 6.1.1 on OSX 10.10.4 does not build Kodi, suggest to recommend XCode 6.3.1 on 10.10

HofiOne committed:
Fixed OS X i386 Xcode builds on x86_64 build machines

Without assigning App-OSX.xcconfig to Kodi.app target building i386
target will not fix dependent libraries when bundling the Kodi.app with
copyframeworks-osx.command

copyframeworks-osx.command uses XBMC_DEPENDS that composed like
$(XBMC_DEPENDS_ROOT)/$(SDK_NAME)_$(CURRENT_ARCH)-target

CURRENT_ARCH not set correctly, without assigning App-OSX.xcconfig the
ONLY_ACTIVE_ARCH is not set and the builder machine architecture will
be used instead of the targeted architecture.

HofiOne commented:
Sorry the first line

+On Mavericks (OSX 10.9.x) we recommend using Xcode 6.3.1.

was accidental, I've tested on 10.10 only and only that change suggested.

if you did it on a stock os and worked that is a much better test.
unfortunately using XCode 6.1.1 is problematic on my boxes (both at home and work) I'd be curious what could cause that problem 

apart from that, i'v made a 2nd big mistake and pushed the 2 changes together when they are not related at all. the change in the xcode project file is necessary to build  correct bundled app from the IDE.

Memphiz commented:
Yes just change your branch so that it only contains the 2 lines in xcode and commit it as extra commit. After that do git rebase HEAD~2 -i and "r"ename the first and "f"ixup the scond commit. After that name the first commit correctly like you did already. After that force push to your repo and this PR gets updated automagically ...

HofiOne committed:
Fixed OS X i386 Xcode builds on x86_64 build machines - Run2

Without assigning App-OSX.xcconfig to Kodi.app target building i386
target will not fix dependent libraries when bundling the Kodi.app with
copyframeworks-osx.command

copyframeworks-osx.command uses XBMC_DEPENDS that composed like
$(XBMC_DEPENDS_ROOT)/$(SDK_NAME)_$(CURRENT_ARCH)-target

CURRENT_ARCH not set correctly, without assigning App-OSX.xcconfig the
ONLY_ACTIVE_ARCH is not set and the builder machine architecture will
be used instead of the targeted architecture.

HofiOne committed:
Merge branch 'Fix-Enhance-XCode-Builds' of https://github.com/HofiOne/xbmc into Fix-Enhance-XCode-Builds

* 'Fix-Enhance-XCode-Builds' of https://github.com/HofiOne/xbmc:
  Fixed OS X i386 Xcode builds on x86_64 build machines

HofiOne committed:
Revert "Fixed OS X i386 Xcode builds on x86_64 build machines"

This reverts commit 24a1d9127fb03af5b5241e300996a00f6b24c21a.

HofiOne commented:
I've successfully f@cked up the revert so creating a new pull request instead, sorry :/
